xcdat/temporal.py

```diff
@@ -1037,7 +1034,12 @@ def _group_data(self, data_var: xr.DataArray) -> DataArrayGroupBy:
         if self._mode == "average":
             dv_gb = dv.groupby(f"{self._dim}.{self._freq}")
         else:
-            self._labeled_time = self._label_time_coords(dv[self._dim])
+            # Reuse the labeled time coordinates if they are already generated.
+            # This specifically applies when generating weights, which calls
+            # this method again after it has already been called to group
+            # the data variable with the same time coordinates.
+            if not hasattr(self, "_labeled_time"):
+                self._labeled_time = self._label_time_coords(dv[self._dim])
```

Comment:
In one of your previous comments, you mentioned that the `self._labeled_time` attrs were being dropped.

This was because we call `_group_data()` twice, one for `dv` and one for `weights`. 
`weights` doesn't preserve the time dimension attrs because it is unnecessary. However, this caused a problem where the second call to `_group_data()` with `weights` overwrites `self._labeled_time` (resulting in attrs being dropped).

This new conditional fixes resolves this problem.